File: exam/src/validator/documentValidators/ContractValidator.java
package validator.documentValidators;

import document.Document;
import validator.DocumentValidator;

public class ContractValidator extends DocumentValidator {

    public ContractValidator() {
        super();
    }

    @Override
    protected boolean validateSize(Document document) {
        return document.getSize() <= 3;
    }

    @Override
    protected boolean validateExtension(Document document) {
        return isPdf(document);
    }

    private boolean isPdf(Document document) {
        return document.getExtension().equals("pdf");
    }

    @Override
    protected boolean validateMetadata(Document document) {
        if (!metadataContainsAuthor(document)) return false;
        if (!metadataContainsVersion(document)) return false;
        return true;
    }

    private boolean metadataContainsAuthor(Document document) {
        return document.getMetadata().containsKey("author");
    }

    private boolean metadataContainsVersion(Document document) {
        return document.getMetadata().containsKey("version");
    }
}

File: exam/src/strategies/validation/ContractValidator.ts
import { DocumentValidator } from "./DocumentValidator";

export class ContractValidator extends DocumentValidator {
    private readonly CONTRACT_MAX_SIZE = 3;
    private readonly CONTRACT_PDF_EXTENSION = '.pdf';
    
    public validateSize(size: number): boolean {
        return super.validateSize(size) && size <= this.CONTRACT_MAX_SIZE;
    }

    public validateExtension(extension: string): boolean {
        return extension == this.CONTRACT_PDF_EXTENSION;
    }

    public validateMetadata(metadata: Record<string, string>): boolean {
        if (metadata["author"] == undefined) {
            return false;
        }

        if (metadata["version"] == undefined) {
            return false;
        }

        return metadata["author"].length > this.DOCUMENT_MIN_NAME_LENGTH && metadata["version"].length > this.DOCUMENT_MIN_NAME_LENGTH;
    }
}